algebird-core/src/main/scala/com/twitter/algebird/Operators.scala

```diff
@@ -43,6 +43,9 @@ class RichTraversable[T](t: TraversableOnce[T]) {
   def sumByKey[K, V](implicit ev: <:<[T, (K, V)], sg: Semigroup[V]): Map[K, V] =
     MapAlgebra.sumByKey(t.map { _.asInstanceOf[(K, V)] })
```

Comment:
these is no reason for the map here. Just cast t:
```scala
MapAlgebra.sumByKey(t.asInstanceOf[TraversableOnce(K, V)])
```
Not sure why that map is there. Same below. This is safe because due to the `ev` we know `T <: (K, V)` and `TraversableOnce` is covariant (so it is safe to upcast the type parameter).

We really need a function like:
```scala
def upcast[T[+_], A, B](t: T[A])(implicit ev: A <:< B): T[B] =
  // it is safe to upcast covariant types:
  t.asInstanceOf[T[B]]
```
```
scala> val t: List[Any] = upcast(List[Int](1, 2))
t: List[Any] = List(1, 2)
scala> val t: List[Int] = upcast(List[Any](1, 2))

<console>:12: error: type mismatch;
 found   : List[Any]
 required: List[Int]
       val t: List[Int] = upcast(List[Any](1, 2))
```